otlp/src/otlp/logs/component.cpp

```diff
@@ -51,19 +52,43 @@ LoggerComponent::LoggerComponent(const components::ComponentConfig& config,
   logger_config.attributes_mapping =
       config["attributes-mapping"]
           .As<std::unordered_map<std::string, std::string>>({});
+  logger_config.logs_sink =
+      config["sinks"]["logs"].As<SinkType>(SinkType::kOtlp);
+  logger_config.tracing_sink =
+      config["sinks"]["tracing"].As<SinkType>(SinkType::kOtlp);
+
   logger_ = std::make_shared<Logger>(std::move(client), std::move(trace_client),
                                      std::move(logger_config));
-
   // We must init after the default logger is initialized
   auto& logging_component = context.FindComponent<components::Logging>();
-  if (logging_component.GetLoggerOptional("default")) {
-    throw std::runtime_error(
-        "You have registered both 'otlp-logger' component and 'default' logger "
-        "in 'logging' component. Either disable default logger or otlp "
-        "logger.");
+  logging::LoggerPtr default_logger{};
+  if (logger_config.logs_sink == SinkType::kOtlp &&
+      logger_config.tracing_sink == SinkType::kOtlp) {
+    if (logging_component.GetLoggerOptional("default")) {
+      throw std::runtime_error(
+          "You've registered both the 'otlp-logger' component and the "
+          "'default' logger in 'logging' component, but have opted to "
+          "send both logs and traces using oltp. Either disable default "
+          "logger or otlp logger.");
+    }
+  } else {
+    try {
+      default_logger = logging_component.GetLogger("default");
+    } catch (const std::exception&) {
```

Comment:
not any exception, but not found logger exception type

Reply:
Here is implementation of  logging component's GetLogger:
```
logging::LoggerPtr Logging::GetLogger(const std::string& name) {
  auto it = loggers_.find(name);
  if (it == loggers_.end()) {
    auto logger = extra_loggers_.Get(name);
    if (logger) return *logger;

    throw std::runtime_error("logger '" + name + "' not found");
  }
  return it->second;
}
```

so what should I catch?